--- Makefile.orig	2009-02-02 18:22:19.000000000 -0500
+++ Makefile	2009-02-02 18:24:20.000000000 -0500
@@ -2,7 +2,7 @@
 ###    This Makefile assumes CASA+CASACore enviroment. 
 ###    For who wants to build ASAP with older CASA without CASACore 
 ###    environment, replace $(CASAPATH) with $(AIPSPATH). 
-PREFIX := /opt/casa/darwin
+PREFIX := __BUILD_ROOT__/linux
 #PREFIX := /export/home/minor/casa/active/linux_gnu
 PYDIR := $(PREFIX)/python/2.5
 
--- src/Makefile.orig	2009-02-02 18:22:19.000000000 -0500
+++ src/Makefile	2009-02-02 19:13:48.000000000 -0500
@@ -11,7 +11,7 @@
 # the casa environment AIPSPATH has to be defined
 CASAROOT  := $(word 1, $(CASAPATH))
 CASAARCH  := $(word 2, $(CASAPATH))
-CASAINC   := -I$(CASAROOT)/code/include -I$(CASAROOT)/$(CASAARCH)/include/casacore -I$(CASAROOT)/$(CASAARCH)/include
+CASAINC   := -I$(CASAROOT)/code/include -I/usr/include/casacore -I$(CASAROOT)/$(CASAARCH)/include
 CASALIB   := $(CASAROOT)/$(CASAARCH)/lib
 THIRDPARTY = /opt/casa/core2-apple-darwin8/3rd-party
 
@@ -22,8 +22,8 @@
 endif
 
 # compiler flags
-CXXFLAGS := -fPIC -O2 -g -DCASA_USECASAPATH
-CXXFLAGS += -ansi -Wall
+CXXFLAGS := -fPIC -O2 -DCASA_USECASAPATH
+CXXFLAGS += -ansi -Wall -D__x86_64__ -DAIPS_64B
 
 # darwin specific flags
 ifeq "$(CASAARCH)" "darwin"
@@ -31,30 +31,26 @@
 endif
 
 # cfitsio
-CFITSIOROOT := $(THIRDPARTY)
+CFITSIOROOT := /usr
 CFITSIOINC := -I$(CFITSIOROOT)/include/cfitsio
 #CFITSIOLIB := -L$(CFITSIOROOT)/lib -lcfitsio
 CFITSIOLIB := -lcfitsio
 
 #rpfits
-RPFITSROOT := $(THIRDPARTY)
+RPFITSROOT :=
 #RPFITSLIB := $(RPFITSROOT)/lib/librpfits.a
 RPFITSLIB := -lrpfits
 
 G2CROOT := /usr
-G2CARCH := $(G2CROOT)/lib/gcc/i386-apple-darwin8.7.1/4.2.0/libgcc.a
-G2CARCH := $(G2CROOT)/lib/gcc/powerpc-apple-darwin8.7.0/4.2.0/libgcc.a
-G2CARCH := $(G2CROOT)/lib/gcc/i386-redhat-linux/4.1.0/libgcc.a
+#G2CARCH := $(G2CROOT)/lib/gcc/i386-apple-darwin8.7.1/4.2.0/libgcc.a
+#G2CARCH := $(G2CROOT)/lib/gcc/powerpc-apple-darwin8.7.0/4.2.0/libgcc.a
+#G2CARCH := $(G2CROOT)/lib/gcc/i386-redhat-linux/4.1.0/libgcc.a
 #G2CLIB := $(G2CROOT)/lib/libgfortran.a
 G2CLIB := -lgfortran
 
 # This assumes all casa libs are static only (*.a)
 # if not than there might by symbol resolution errors.
-CASAPPLIB := -L$(CASALIB) \
-             -latnf -lcasa_images -lcasa_ms -lcasa_components -lcasa_coordinates \
-             -lcasa_lattices -lcasa_fits -lcasa_measures -lcasa_measures_f \
-             -lcasa_tables -lcasa_scimath -lcasa_scimath_f -lcasa_casa  \
-             $(CASALIB)/libwcs.a \
+CASAPPLIB := -L$(CASALIB) -lcasacore -latnf -lwcs \
              $(RPFITSLIB) $(CFITSIOLIB) $(G2CLIB) -lstdc++
 
 # darwin specific CASA flags
@@ -64,9 +60,17 @@
    CASAPPLIB += -llapack -lblas
 endif
 
+# the linker
+LD        := $(CXX)
+
+LDFLAGS   := -shared -Wl$(TARGET) -s
+ifeq "$(CASAARCH)" "darwin"
+   LDFLAGS   := -bundle
+endif
+
 # python
 PYVERSION := 2.5
-PYTHONROOT := /usr/lib/casapy
+PYTHONROOT := /usr/lib64/casapy
 PYTHONINC := -I$(PYTHONROOT)/include/python$(PYVERSION)
 PYTHONLIB := -L$(PYTHONROOT)/lib -lpython$(PYVERSION)
 
@@ -83,13 +87,12 @@
 
 # has to be build with same g++ version as casa
 #BOOSTROOT := /home/casa/boost
-BOOSTROOT := $(THIRDPARTY)
-#BOOSTLIB  := -L$(BOOSTROOT)/lib -lboost_python-gcc34
-BOOSTLIB  := -L$(BOOSTROOT)/lib -lboost_python-mt-1_35
-BOOSTINC  := -I$(BOOSTROOT)/include/boost-1_35
+BOOSTROOT := /usr/lib64/casapy
+BOOSTLIB  := -L$(BOOSTROOT)/lib -lboost_python
+BOOSTINC  := -I$(BOOSTROOT)/include
 #BOOSTINC  := -I$(BOOSTROOT)/include/boost
 
-LDFLAGS   := -shared -Wl$(TARGET) -s -Xlinker -rpath -Xlinker $(BOOSTROOT)/lib:$(PYTHONROOT)/lib:$(CASALIB)
+LDFLAGS   := -shared -Wl$(TARGET) -s -Xlinker -rpath -Xlinker $(BOOSTROOT)/lib:$(PYTHONROOT)/lib
 ifeq "$(CASAARCH)" "darwin"
    LDFLAGS   := -bundle
 endif
